model/src/test/java/com/javahelp/model/userEntityTest/UserPasswordTest.java

```diff
@@ -0,0 +1,160 @@
+package com.javahelp.model.userEntityTest;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import com.javahelp.model.user.UserInfo;
+import com.javahelp.model.user.ClientUserInfo;
+import com.javahelp.model.user.UserType;
+import com.javahelp.model.user.ProviderUserInfo;
+
+import java.util.Base64;
+import java.util.Random;
+
+public class UserPasswordTest {
+    /** A testing file for UserPassword entity, where:
+     *
+     * --- Attributes ---
+     * rand - a variable of Random type to generate random byte[] representation of an input.
+     * saltOne - the current salt for the first user
+     * saltTwo - the current salt for the second user
+     * passWord1 - the current given password of the first user
+     * passWord2 - the current given password of the second user
+     *
+     * user1Pass - an object represents for the first user's password
+     * user2Pass - an object represents for the second user's password
+     * user1 - the first user
+     * user2 - the second user
+     */
+
+    private Random rand;
```

Comment:
The `Random` instance does not really need to be stored in the class here

Reply:
Also just a note for the future, you don't need to put `git stage . && git commit -m "commit message" && git push` in your commit messages. It seems like you might be typing this into the message space in Android Studio, but these are console commands. The message space is just for the commit message, and Android Studio does the rest automatically.

Reply:
Oh I see, I'll do ``git commit -m "this is a commit message"`` next time

Reply:
You can just write the commit message in the commit message box in Android Studio and it'll do all the command line stuff automatically. I wouldn't worry about any git commands.